src/main/java/org/fungover/Topic.java

```diff
@@ -0,0 +1,29 @@
+package org.fungover;
+
+import java.util.Objects;
+
+public class Topic {
+    String name;
+
+    private Topic (String name) {
+        this.name = name;
```

Comment:
Consider making the "name" field private. This promotes encapsulation and allows better control over how the field is accessed.

Since it doesn't seem like you intend to modify the "name" after it's been initalized maybe you could make it final? This guartantees that the variabel cannot be reassigned.

Reply:
Thank you for the feedback, we made the "name" fileld private and added getName!